### PR TITLE
callstack: deprecate all classes that were moved to core

### PR DESCRIPTION
**Pull request by arfio**
Deprecate all the classes from callstack and analysis that were moved to core. Change the classes that were not moved to depend on profiling.core and profiling.core.tests.
Dependent on https://github.com/eclipse-tracecompass-incubator/org.eclipse.tracecompass.incubator/pull/18 that will remove every dependency to these classes

### CODE REVIEW

**Review by bhufmann — CHANGES_REQUESTED** [review]
The Trace Compass incubator project doesn't provide any releases, since it's an incubator project. However, we update the version number whenever Trace Compass mainline is released, just to have some tracking (even if it's not a release :-)). 

So, currently it's at 0.9.0 and you will have to update all deprecated annotations to since="0.10.0".  9.4 is the Trace Compass mainline release and can't be used here. I commented only on 2 representative APIs to change. This comment is valid for all the others.

I noticed that the build is failing, because other plug-ins using deprecated APIs. This are settings that we chose for this project. You, could either change the setting or change the implementation to use the Trace Compass mainline APIs instead. I prefer the latter.

File: callstack/org.eclipse.tracecompass.incubator.analysis.core/src/org/eclipse/tracecompass/incubator/analysis/core/concepts/ICallStackSymbol.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 /**
  * @author Geneviève Bastien
  */
+@Deprecated(since="9.4", forRemoval=true)
```
[the remark this answers is not in the document]
Comment (arfio) [reply callstack/org.eclipse.tracecompass.incubator.analysis.core/src/org/eclipse/tracecompass/incubator/analysis/core/concepts/ICallStackSymbol.java]: Done.

File: callstack/org.eclipse.tracecompass.incubator.analysis.core/src/org/eclipse/tracecompass/incubator/analysis/core/concepts/AggregatedCallSite.java [hunk 2]
```diff
@@ -27,6 +27,7 @@
  *
  * @author Geneviève Bastien
  */
+@Deprecated(since="9.4", forRemoval=true)
```
[the remark this answers is not in the document]
Comment (arfio) [reply callstack/org.eclipse.tracecompass.incubator.analysis.core/src/org/eclipse/tracecompass/incubator/analysis/core/concepts/AggregatedCallSite.java]: Done.

**Comment by arfio** [comment]
Thank you for the review, I have changed the deprecation version argument like you suggested. For the other plug-ins using the deprecated API, I have another pull request opened for this: https://github.com/eclipse-tracecompass-incubator/org.eclipse.tracecompass.incubator/pull/18

**Review by bhufmann — CHANGES_REQUESTED** [review]
To fix the build, change the settings for using deprecated APIs in ROCm plug-in to warning:
https://github.com/eclipse-tracecompass-incubator/org.eclipse.tracecompass.incubator/blob/ddb25ba7240ef442c0571706f1a4eb6e319cba37/analyses/org.eclipse.tracecompass.incubator.rocm.core/.settings/org.eclipse.jdt.core.prefs#L35

**Review by bhufmann — APPROVED** [review]
Thanks for this contribution. Looks good to me.

**Pull request merged**